common/types.go

```diff
@@ -802,3 +803,270 @@ func (b *BuilderSubmitBlockRequest) Withdrawals() []*consensuscapella.Withdrawal
 	}
 	return nil
 }
+
+/*
+SubmitBlockRequestV2Optimistic is the v2 request from the builder to submit
+a block. The message must be SSZ encoded. The first three fields are at most
+944 bytes, which fit into a single 1500 MTU ethernet packet. The
+`UnmarshalSSZHeaderOnly` function just parses the first three fields,
+which is sufficient data to set the bid of the builder. The `Transactions`
+and `Withdrawals` fields are required to construct the full SignedBeaconBlock
+and are parsed asynchronously.
+
+Header only layout:
+[000-236) = Message   (236 bytes)
+[236-240) = offset1   (  4 bytes)
+[240-336) = Signature ( 96 bytes)
+[336-340) = offset2   (  4 bytes)
+[340-344) = offset3   (  4 bytes)
+[344-944) = EPH       (600 bytes)
+*/
+type SubmitBlockRequestV2Optimistic struct {
+	Message                *apiv1.BidTrace
+	ExecutionPayloadHeader *consensuscapella.ExecutionPayloadHeader
```

Comment:
what's the upgrade path for deneb?

Reply:
the deneb blocks will have the kzg commitments. The ExecutionPayload and the ExecutionPayloadHeader each have two new fields, `blob_gas_used` and `excess_blob_gas` for sure will need to include these types because they are part of what the validator signs over.

Reply:
are you making a versioned struct for future upgrades?